src/commands/publish.rs

```diff
@@ -58,9 +50,12 @@ where
 }
 
 /// Submits a publish to the registry.
-async fn submit(client: &FileSystemClient, info: PublishInfo) -> Result<()> {
-    let signing_key = demo_signing_key()?;
-    client.publish_with_info(&signing_key, info).await?;
+async fn submit(
+    client: &FileSystemClient,
+    info: PublishInfo,
+    signing_key: &PrivateKey,
```

Comment:
Instead of taking the signing key here, if we instead pass in the username, we could call

```rust
get_signing_key(user, host)
```

where `host` could be derived from `client.url()`?

That would eliminate the need for `CommonOptions::get_signing_key`.

It would also delay any interactions with the key store until right before we know we're submitting a publish operation.

Reply:
@Kylebrown9 the user name associated with the signing-key (correlated via `warg key add ...`)

Reply:
`cargo-component` uses the same strategy so that one signing key isn't used per-registry, but the CLI user gets to decide which key to use for that registry to publish with.

Perhaps a better name is needed? Like "key name" or some such? I agree that "user name" could be confused with some sort of auth for the registry itself.

Reply:
`--key-name` (or `--key`?) sounds good to me; @Kylebrown9 do you have a preference?

Reply:
I updated this PR to use `--key-name` instead of `--user`.